Sources/Sentry/include/NSURLSessionTask+Sentry.m

```diff
@@ -0,0 +1,25 @@
+#import "NSURLSessionTask+Sentry.h"
+
+
+@implementation
+NSURLSessionTask (Sentry)
+
+- (nullable NSString *)sentry_graphQLOperationName
+{
+    if (!self.originalRequest.HTTPBody) { return nil; }
+    if (![[self.originalRequest valueForHTTPHeaderField:@"Content-Type"] isEqual: @"application/json"]) { return nil; }
+
+    NSError *error = nil;
+    id requestDictionary = [NSJSONSerialization JSONObjectWithData:self.originalRequest.HTTPBody options:0 error:&error];
+
+    if (error) { return nil; }
+    if (![requestDictionary isKindOfClass: [NSDictionary class]]) { return nil; } // Could be an array
+
+    id operationName = [requestDictionary valueForKey:@"operationName"];
+    if (![operationName isKindOfClass: [NSString class]]) { return nil; }
+    if ([operationName length] == 0) { return nil; }
+
+    return operationName;
+}
```

Comment:
We can't use Objc Categories to extend a class because our static lib will strip it. We need to turn this into a C function.

Reply:
Cheers @brustolin, I couldn't see an example of how this has been done before so I have changed the function to be like other ones in nearby files. If this still won't work, please let me know where I can find an example of a C-style helper function!

Reply:
This works fine. Thanks!!